components/brave_wallet/browser/json_rpc_service.cc

```diff
@@ -85,6 +92,29 @@ net::NetworkTrafficAnnotationTag GetNetworkTrafficAnnotationTag() {
     )");
 }
 
+// https://tools.ietf.org/html/rfc1035#section-3.1
+absl::optional<std::vector<uint8_t>> DnsEncode(const std::string& dotted_name) {
+  std::vector<uint8_t> result;
+  result.resize(dotted_name.size() + 2);
+  result.front() = '.';  // Placeholder for first label length.
+  base::ranges::copy(dotted_name, result.begin() + 1);
+  result.back() = '.';  // Placeholder for terminal zero byte.
+
+  size_t last_dot_pos = 0;
+  for (size_t i = 1; i < result.size(); ++i) {
+    if (result[i] == '.') {
+      size_t label_len = i - last_dot_pos - 1;
```

Comment:
this math assigning a possibly signed to unsigned seems would be better with an extra check.
Did we tested for UTF-8 strings anyway?

Reply:
This math is all unsigned
Will add [similar](https://source.chromium.org/chromium/chromium/src/+/main:net/dns/dns_util.cc;drc=bead2b4d2553e57229747b48b3dceaadbb8972bf;l=142?q=DNSDomainFromDot&ss=chromium) byte validation. Thanks!

Reply:
Not sure if it is the correct validation over there. https://docs.ens.domains/frequently-asked-questions `nameprep` seems the standard to follow here, to validate these names. Regarding these, we should be careful in considering multibyte runes when iterating through characters instead of single-byte sequences.

Reply:
Discussed in slack: we already support only ascii domain names for ENS, so no nameprep here. Will add explicit byte validation to DnsEncode.

Reply:
https://github.com/brave/brave-core/pull/14447/commits/b398bac78f1c7e868d6e49ab6aa9fc0be5ac4c37